Snowflake.Data.Tests/UnitTests/StructuredTypesTest.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Snowflake.Data.Core.Converter;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    [TestFixture]
+    public class StructuredTypesTest
+    {
+        [Test]
+        [TestCaseSource(nameof(TimeConversionCases))]
+        public void TestTimeConversions(string value, string sfTypeString, object expected)
+        {
+            // arrange
+            var timeConverter = new TimeConverter();
+            var sfType = (SFTimestampType) Enum.Parse(typeof(SFTimestampType), sfTypeString);
+            var csharpType = expected.GetType();
+
+            // act
+            var result = timeConverter.Convert(value, sfType, csharpType);
+
+            // assert
+            Assert.AreEqual(expected, result);
+        }
+
+        internal static IEnumerable<object[]> TimeConversionCases()
+        {
+            yield return new object[] {"2024-07-11 14:20:05", SFTimestampType.TIMESTAMP_NTZ.ToString(), DateTime.Parse("2024-07-11 14:20:05").ToUniversalTime()};
```

Comment:
What about millis/micros?

Reply:
this is an unit test. Actually it is duplicated in integration tests, where I test the dates with maximum precision. But I added here tests for precision as well.